resources/js/Components/Layout/AuthOverlay/AuthOverlay.tsx

```diff
@@ -34,12 +35,15 @@ export const AuthOverlay = ({
         switching,
         errorMessage,
         waitingSignature,
-        requiresSignature,
+        requiresSignature: metaMaskRequiresSignature,
+        signed,
     } = useMetaMaskContext();
 
+    const requiresSignature = (mustBeSigned && !signed) || metaMaskRequiresSignature;
+
     const showSignMessage = useMemo(
-        () => requiresSignature && !waitingSignature && errorMessage === undefined && !connecting,
-        [requiresSignature, waitingSignature, errorMessage, connecting],
+        () => metaMaskRequiresSignature && !waitingSignature && errorMessage === undefined && !connecting,
+        [metaMaskRequiresSignature, waitingSignature, errorMessage, connecting],
```

Comment:
`useMemo` is an overkill here, as it will trigger recalculation almost every instance.